horace_core/sqw/@PixelData/PixelData.m

```diff
@@ -0,0 +1,361 @@
+    % Returns the coordinates of the pixels in the projection axes, i.e.: u1,
+    % u2, u3 and dE (4 x n array)
+    coordinates;
+
+    % The run index the pixel originated from (1 x n array)
+    run_idx;
+
+    % The detector group number in the detector listing for the pixels (1 x n array)
+    detector_idx;
+
+    % The energy bin numbers (1 x n array)
+    energy_idx;
+
+    % The signal array (1 x n array)
+    signals;
+
+    % The variance on the signal array (variance i.e. error bar squared) (1 x n array)
+    variance;
+
+    % The number of pixels in the data block
+    num_pixels;
+
+    % Returns the full raw pixel data block usage of this attribute is discouraged, the structure of the return value is not guaranteed
+    data;
+end
+
+methods (Static)
+
+    function obj = cat(varargin)
+        % Concatentate the given PixelData objects' pixels. This function performs
+        % a straight-forward data concatenation.
+        %
+        %   >> joined_pix = PixelData.cat(pix_data1, pix_data2);
+        %
+        % Input:
+        % ------
+        %   varargin    A cell array of PixelData objects
+        %
+        % Output:
+        % -------
+        %   obj         A PixelData object containing all the pixels in the inputted
+        %               PixelData objects
+        data_cell_array = cellfun(@(p) p.data, varargin, 'UniformOutput', false);
+        data = cat(2, data_cell_array{:});
+        obj = PixelData(data);
+    end
+
+    function obj = loadobj(S)
+        % Load a PixelData object from a .mat file
+        %
+        %   >> obj = loadobj(S)
+        %
+        % Input:
+        % ------
+        %   S       An instance of this object
+        %
+        % Output:
+        % -------
+        %   obj     An instance of this object
+
+        obj = PixelData(S);
+    end
+
+end
+
+methods
+
+    function obj = PixelData(data)
+        % Construct a PixelData object from the given data. Default
+        % construction initialises the underlying data as an empty (9 x 0)
+        % array.
+        %
+        %   >> obj = PixelData(ones(9, 200))
+        %
+        %   >> obj = PixelData(200)  % intialise 200 pixels with underlying data set to zero
+        %
+        % Input:
+        % ------
+        %   data    A 9 x n matrix, where each row corresponds to a pixel and
+        %           the columns correspond to the following:
+        %             col 1: u1
+        %             col 2: u2
+        %             col 3: u3
+        %             col 4: u4
+        %             col 5: run_idx
+        %             col 6: detector_idx
+        %             col 7: energy_idx
+        %             col 8: signals
+        %             col 9: variance
+        %
+        %  data    An integer specifying the desired number of pixels. The underlying
+        %          data will be filled with zeros
+        %
+        if nargin == 1
+            if isa(data, 'PixelData')
+                obj.data = data.data;
```

Comment:
`data` as an argument name is confusing given the later use of `data.data` as we have a `data` attribute